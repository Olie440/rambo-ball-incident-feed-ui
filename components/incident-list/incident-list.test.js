import IncidentList, { Header } from './incident-list.component';
import { shallow } from 'enzyme';

describe('<IncidentList />', () => {
    it('passes the correct props to PaginatedList', () => {
        const props = {
            pageSize: 25,
            incidents: [{id: 1}]
        };

        const element = shallow(<IncidentList {...props} />);
        expect(element).toMatchSnapshot();
    });

    it('doesn\'t error when no props are passed in', () => {
        const element = shallow(<IncidentList  />);
        expect(element).toMatchSnapshot();
    });

    describe('<Header />', () => {
        it('renders the correct HTML', () => {
            const element = shallow(<Header />);
            expect(element).toMatchSnapshot();
        });
    });
});