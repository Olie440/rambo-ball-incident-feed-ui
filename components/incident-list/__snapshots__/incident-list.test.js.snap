// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncidentList /> <Header /> renders the correct HTML 1`] = `
<header
  className="header"
>
  <span>
    Time
  </span>
  <span>
    Incident Type
  </span>
</header>
`;

exports[`<IncidentList /> doesn't error when no props are passed in 1`] = `
<PaginatedList
  classNames={
    Object {
      "container": "incidentList",
      "controls": Object {
        "container": "pageControls",
        "leftButton": "pageButton",
        "rightButton": "pageButton",
      },
    }
  }
  component={[Function]}
  data={Array []}
  header={[Function]}
  pageSize={20}
/>
`;

exports[`<IncidentList /> passes the correct props to PaginatedList 1`] = `
<PaginatedList
  classNames={
    Object {
      "container": "incidentList",
      "controls": Object {
        "container": "pageControls",
        "leftButton": "pageButton",
        "rightButton": "pageButton",
      },
    }
  }
  component={[Function]}
  data={
    Array [
      Object {
        "id": 1,
      },
    ]
  }
  header={[Function]}
  pageSize={25}
/>
`;
